indico/migrations/versions/20191023_1820_2496c4adc7e9_add_track_principal.py

```diff
@@ -0,0 +1,304 @@
+        sa.Column('permissions', postgresql.ARRAY(sa.String()), nullable=False),
+        sa.Column('id', sa.Integer(), nullable=False),
+        sa.Column('track_id', sa.Integer(), nullable=False, index=True),
+        sa.Column('local_group_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('mp_group_provider', sa.String(), nullable=True),
+        sa.Column('event_role_id', sa.Integer(), nullable=True, index=True),
+        sa.Column('mp_group_name', sa.String(), nullable=True),
+        sa.Column('user_id', sa.Integer(), nullable=True, index=True),
+        sa.Column(
+            'type',
+            PyIntEnum(PrincipalType, exclude_values={PrincipalType.network}),
+            nullable=False,
+        ),
+        sa.Column('email', sa.String(), nullable=True, index=True),
+        sa.CheckConstraint(
+            u'NOT full_access', name=op.f('ck_track_principals_no_full_access')
+        ),
+        sa.CheckConstraint(
+            u'NOT read_access', name=op.f('ck_track_principals_no_read_access')
+        ),
+        sa.CheckConstraint(
+            u'email IS NULL OR email = lower(email)',
+            name=op.f('ck_track_principals_lowercase_email'),
+        ),
+        sa.CheckConstraint(
+            u'read_access OR full_access OR array_length(permissions, 1) IS NOT NULL',
+            name=op.f('ck_track_principals_has_privs'),
+        ),
+        sa.CheckConstraint(
+            u'type != 1 OR (email IS NULL AND event_role_id IS NULL AND local_group_id IS NULL AND '
+            'mp_group_name IS NULL AND mp_group_provider IS NULL AND user_id IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_user'),
+        ),
+        sa.CheckConstraint(
+            u'type != 2 OR (email IS NULL AND event_role_id IS NULL AND mp_group_name IS NULL AND '
+            'mp_group_provider IS NULL AND user_id IS NULL AND local_group_id IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_local_group'),
+        ),
+        sa.CheckConstraint(
+            u'type != 3 OR (email IS NULL AND event_role_id IS NULL AND local_group_id IS NULL AND '
+            'user_id IS NULL AND mp_group_name IS NOT NULL AND mp_group_provider IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_multipass_group'),
+        ),
+        sa.CheckConstraint(
+            u'type != 4 OR (event_role_id IS NULL AND local_group_id IS NULL AND mp_group_name IS NULL AND '
+            'mp_group_provider IS NULL AND user_id IS NULL AND email IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_email'),
+        ),
+        sa.CheckConstraint(
+            u'type != 6 OR (email IS NULL AND local_group_id IS NULL AND mp_group_name IS NULL AND '
+            'mp_group_provider IS NULL AND user_id IS NULL AND event_role_id IS NOT NULL)',
+            name=op.f('ck_track_principals_valid_event_role'),
+        ),
+        sa.ForeignKeyConstraint(
+            ['event_role_id'],
+            [u'events.roles.id'],
```

Comment:
why `u` here? does it have to be unicode? I don't think we use `unicode` strings in other places

Reply:
this is autogenerated, I should have taken a closer look at that :) thanks for pointing out